Cython/Compiler/FlowControl.py

```diff
@@ -590,7 +590,7 @@ def check_definitions(flow, compiler_directives):
             if (node.allow_null or entry.from_closure
                     or entry.is_pyclass_attr or entry.type.is_error):
                 pass  # Can be uninitialized here
-            elif node.cf_is_null:
+            elif node.cf_is_null and not entry.in_closure:
```

Comment:
This is the fix for comment https://github.com/cython/cython/issues/3699#issuecomment-647134169 (but not the main issue) - I suspect you could do better in principle (e.g. work out if the closure is actually called), however it's hard to reason much about entries in closures.